internal/api/v1/service/match.go

```diff
@@ -0,0 +1,60 @@
+package service
+
+import (
+	"strings"
+
+	"github.com/epinio/epinio/helpers/kubernetes"
+	"github.com/epinio/epinio/internal/api/v1/response"
+	"github.com/epinio/epinio/internal/cli/server/requestctx"
+	"github.com/epinio/epinio/internal/services"
+	apierror "github.com/epinio/epinio/pkg/api/core/v1/errors"
+	"github.com/epinio/epinio/pkg/api/core/v1/models"
+
+	"github.com/gin-gonic/gin"
+)
+
+// Match handles the API endpoint /namespace/:namespace/servicesmatches/:pattern (GET)
+// It returns a list of all Epinio-controlled services matching the prefix pattern.
+func (oc Controller) Match(c *gin.Context) apierror.APIErrors {
+	ctx := c.Request.Context()
+	log := requestctx.Logger(ctx)
+
+	namespace := c.Param("namespace")
+
+	log.Info("match services")
+	defer log.Info("return")
+
+	cluster, err := kubernetes.GetCluster(ctx)
+	if err != nil {
+		return apierror.InternalError(err)
+	}
+
+	log.Info("list services")
+	kubeServiceClient, err := services.NewKubernetesServiceClient(cluster)
+	if err != nil {
+		return apierror.InternalError(err)
+	}
+
+	serviceList, err := kubeServiceClient.ListInNamespace(ctx, namespace)
```

Comment:
This is a "namespaced" resource. The user will get a 403 if the namespace is not accessible when this is merged: https://github.com/epinio/epinio/pull/1490 . No need to filter anything since this code won't even run if the user doesn't have access to this namespace. Maybe we can add a test to check the 403 response though (after rebasing on #1490).